azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -1417,6 +1503,21 @@ func expandVirtualMachineScaleSetSku(d *schema.ResourceData) (*compute.Sku, erro
 	return sku, nil
 }
 
+func expandAzureRmRollingUpgradePolicy(d *schema.ResourceData) *compute.RollingUpgradePolicy {
+	if configs, ok := d.Get("rolling_upgrade_policy").([]interface{}); ok && len(configs) > 0 {
+		config := configs[0].(map[string]interface{})
+
+		return &compute.RollingUpgradePolicy{
+			MaxBatchInstancePercent:             utils.Int32(int32(config["max_batch_instance_percent"].(int))),
+			MaxUnhealthyInstancePercent:         utils.Int32(int32(config["max_unhealthy_instance_percent"].(int))),
+			MaxUnhealthyUpgradedInstancePercent: utils.Int32(int32(config["max_unhealthy_upgraded_instance_percent"].(int))),
+			PauseTimeBetweenBatches:             utils.String(config["pause_time_between_batches"].(string)),
+		}
+	} else {
```

Comment:
this else is redundant, above if statement returns a value.

Reply:
@kt@katbyte.me is right. Please take at this link:
[https://golang.org/ref/spec#Return_statements](https://golang.org/ref/spec#Return_statements)

The return values will be initialized to 0 while entering into function, so return `nil` here can be saved.






---
In reply to: [208484494](https://github.com/terraform-providers/terraform-provider-azurerm/pull/922#discussion_r208484494) [](ancestors = 208484494)